### PR TITLE
Add length check and space padding to CHAR type

### PR DESCRIPTION
**Pull request by hannes**
This PR adds the fixed-width, space-padded `CHAR(x)` type.

### CODE REVIEW

**Comment by Mause** [comment]
Not adding it to the test_all_types() function yet?

File: test/sql/cast/char.test [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# name: test/sql/cast/char.test
+# description: Issue #2588: Incorrect result with date conversion
+# group: [cast]
+
+statement ok
+PRAGMA enable_verification
+
+query III
+select 'asdf'::CHAR(3), 'asdf'::CHARACTER(3), 'asdf'::BPCHAR(3)
```
Comment (Mytherin) [inline test/sql/cast/char.test:9]: Could we add some unicode tests, e.g.:

```sql
postgres=# select '🦆🦆🦆🦆'::char(3);
 bpchar 
--------
 🦆🦆🦆
```

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks great - one comment:

File: test/sql/cast/char.test (hunk 1 above)
Comment (hannes) [reply test/sql/cast/char.test:9]: oh yeah that's a fun one.

Comment (Mytherin) [reply test/sql/cast/char.test:9]: `LengthFun::IsCharacter` can maybe help here - it figures out if a byte is the first byte in a unicode codepoint.

Comment (hannes) [reply test/sql/cast/char.test:9]: ended up using substring directly

**Comment by hannes** [comment]
@Mause indeed it still should be added to all types

**Comment by hannes** [comment]
I'm not excited about this one perhaps we just decide that those types can go somewhere else

**Pull request closed without merging**